src/Uno.UI.DualScreen/DuoApplicationViewSpanningRect.cs

```diff
@@ -31,91 +32,114 @@ public IReadOnlyList<Rect> GetSpanningRects()
 			{
 				InitializeHelper(currentActivity);
 
-				if (_isDualScreenDevice.Value && _helper.IsDualMode)
+				if (_isDualScreenDevice.Value)
 				{
-					var wuxWindowBounds = ApplicationView.GetForCurrentView().VisibleBounds.LogicalToPhysicalPixels();
-					var helperOrientation = _helper.GetRotation();
-					var wuOrientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
+					if (_helper.IsDualMode)
+					{
+						var helperOrientation = _helper.GetRotation();
+						if (_previousMode.orientation == helperOrientation && _previousMode.result != null)
+						{
+							return _previousMode.result;
+						}
 
-					var occludedRects = _helper
-						.DisplayMask
-						.GetBoundingRectsForRotation(helperOrientation)
-						.Select(r => (Rect)r)
-						.ToArray();
+						_previousMode.orientation = helperOrientation;
+						_previousMode.result = null;
 
-					if (occludedRects.Length > 0)
-					{
-						Rect bounds = wuxWindowBounds;
-						var occludedRect = occludedRects[0];
-						var intersection = bounds;
-						intersection.Intersect(occludedRect);
+						var wuxWindowBounds = ApplicationView.GetForCurrentView().VisibleBounds.LogicalToPhysicalPixels();
+						var wuOrientation = DisplayInformation.GetForCurrentView().CurrentOrientation;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'wuOrientation'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4993239)